planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -481,20 +482,20 @@ void AvoidanceModule::fillShiftLine(AvoidancePlanningData & data, DebugData & de
     data.yield_required = true;
     RCLCPP_WARN_THROTTLE(
       getLogger(), *clock_, 5000,
-      "found safe avoidance path, but it is not safe. canceling avoidance path...");
+      "found avoidance path, but it is not safe. canceling avoidance path...");
   }
 
   /**
    * TODO(Satoshi OTA) Think yield maneuver in the middle of avoidance.
    * Even if it is determined that a yield is necessary, the yield maneuver is not executed
    * if the avoidance has already been initiated.
    */
-  if (!data.safe && data.avoiding_now) {
+  constexpr double THRESHOLD = -0.5;
+  const auto exceed_stop_line = data.to_stop_line < THRESHOLD;
```

Comment:
Just out of curiosity, when you want to determine whether the vehicle has crossed the stop line, wouldn't it be sufficient to use `data.to_stop_line < 0` or `data.to_stop_line < (THRESHOLD=0)`? Is there a reason for allowing a little leeway with `THRESHOLD = -0.5`?

Reply:
@kyoichi-sugahara 

At first the ego try stopping at the stop line (then `to_stop_line = 0`). But sometimes the ego can't stop and exceed it just a bit because of the bad stopping accuracy.

If I set the threshold zero, the module judges the ego already exceeded the stop line and should exit yield maneuver even when actually the ego hasn't stopped yet.

So, I added some buffer for the stopping accuracy in order to prevent this phenomenon.

Reply:
However, on second thought, it seems not to be a good logic. So, I fixed it so that the module check the objects whom the ego stopped for explicitly, and ego exits yield maneuver only when the ego already stoped for the object.

As a result, I could remove this unreadable magic number.